### ISSUE TITLE
Add a message about unsatisfiable constraints

### ISSUE THREAD

**Issue by NathanReb** [avsm/duniverse#260]
We recently introduced messages upon solver failure to warn the users that the solver couldn't find a solution because some packages don't build with dune, listing those packages.

This is great because it is both very clear and actionable for the user, even though writing a dune port won't always be very appealing to users.

In the same fashion we introduced a message that woud disambiguate that even though a package had no dune building version, the main reason why it got rejected is because it had no existing version that match the constraints.

It would be nice to have a generic version of that last message because, similar to the dune thing, it is very clear then why the solver finds no solution and is in most cases very actionable: the user wrote the wrong constraint somewhere or forgot to add a repository or to update one.

To sum up it would be nice if upon failure to find a solution to the dependency specification, we listed the packages that have no satisfying versions available and the packages that have but don't build with dune at all.
